libyara/exec.c

```diff
@@ -1304,10 +1304,28 @@ int yr_execute_code(YR_SCAN_CONTEXT* context)
 
       pop(r2);
 
-      if (is_undef(r2))
-        r1.i = found >= count ? 1 : 0;
-      else
-        r1.i = found >= r2.i ? 1 : 0;
+      if (opcode == OP_OF)
+      {
+        YR_DEBUG_FPRINTF(2, stderr, "- case OP_OF: // %s()\n", __FUNCTION__);
+
+        if (is_undef(r2))
+          r1.i = found >= count ? 1 : 0;
+        else
+          r1.i = found >= r2.i ? 1 : 0;
+      }
+      else  // OP_OF_PERCENT
+      {
+        YR_DEBUG_FPRINTF(
+            2, stderr, "- case OP_OF_PERCENT: // %s()\n", __FUNCTION__);
+
+        // If, by some weird reason, we manage to get an undefined string
+        // reference as the first thing on the stack then count would be zero.
+        // I don't know how this could ever happen but better to check for it.
+        if (count == 0)
```

Comment:
Probably want to also check for "r2.i == YR_UNDEFINED" here too. If we do something like `pe.number_of_sections% of them" and `pe.number_of_sections` is undefined then this will evaluate to true:

```
wxs@wxs-mbp yara % cat rules/test.yara
import "pe"

rule a {
  strings:
    $b = "foo"
  condition:
    pe.number_of_sections% of them
}
wxs@wxs-mbp yara % ./yara -s rules/test.yara /bin/ls
a /bin/ls
wxs@wxs-mbp yara %
```